src/main/java/fruitshop/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() >= 0) {
+            storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
```

Comment:
It's good that you are checking for a non-negative quantity before updating the storage. However, the `BalanceOperation` should not directly manipulate the storage. Consider abstracting this logic into a service layer that interacts with the storage.